tests/test_kms.py

```diff
@@ -268,3 +268,32 @@ def test_kms_key_remove_tag(self):
         self.assertEqual(resources[0]["KeyId"], key_id)
         tags = client.list_resource_tags(KeyId=key_id)["Tags"]
         self.assertEqual(len(tags), 0)
+
+    def test_kms_key_alias(self):
+        session_factory = self.replay_flight_data("test_kms_key_alias")
+        p = self.load_policy(
+            {
+                "name": "dms-instance-kms-key-alias",
+                "resource": 'dms-instance',
+                "filters": [
+                    {
+                        "type": "kms-key-alias",
+                        "key": "c7n:AliasName",
+                        "value": "alias/aws/dms",
+                        "op": "eq"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        client = session_factory().client("kms")
+        resource_kms_key = resources[0]['KmsKeyId']
+        aliases = client.list_aliases(KeyId=resource_kms_key)
+        target_key_arn = None
+        if aliases['Aliases'][0].get('AliasName') == 'alias/aws/dms':
+                target_key_id = aliases['Aliases'][0].get('TargetKeyId')
+                target_key_arn = client.describe_key(
+                    KeyId=target_key_id).get('KeyMetadata').get('Arn')
+        self.assertEqual(len(resources), 1)
```

Comment:
this assert should ideally be before your accessing resources[0] if you got an empty set back.